virtualmachine/aws/wait.go

```diff
@@ -139,3 +156,129 @@ func waitUntilReady(svc *ec2.EC2, instanceID string) error {
 
 	return rerr
 }
+
+// waitUntilStopped: waits until vm is stopped
+func waitUntilStopped(svc *ec2.EC2, instanceID string) error {
+	const maxRetries = VmOpsTimeout / VmOpsInterval
+
+	var instanceStatus *InstanceStatus
+	var err error
+
+	for ii := 0; ii < maxRetries; ii++ {
+		// Sleep will be VmOpsInterval, 2*VmOpsInterval, 3*VmOpsInterval, ...
+		time.Sleep(time.Duration(VmOpsInterval*(ii+1)) * time.Second)
+
+		instanceStatus, err = GetInstanceStatus(svc, instanceID)
+		if err != nil {
+			continue
+		}
+		state := instanceStatus.State
+		switch state {
+		case ec2.InstanceStateNameStopped:
+			return nil
+		case ec2.InstanceStateNameTerminated, ec2.InstanceStateNameShuttingDown:
+			// Polling is useless. This instance is terminated or terminating.
+			// Break the loop.
+			return fmt.Errorf("Instance unexpectedly terminating " +
+				"or has terminated")
+		}
+	}
+
+	return fmt.Errorf("Timeout while waiting for VM to be stopped")
+}
+
+// waitUntilRunning: waits until vm is running
+func waitUntilRunning(svc *ec2.EC2, instanceID string) error {
```

Comment:
Have combined three wait functions, for Start, Stop and Terminate VMs. Have kept CreateVM wait functions as it is because it returns error with details of any missing parameter such as PublicIP, SubnetId etc which are mandatory for VM. So because of this its structure is not same as other three which we have implemented.